Tools/CISupport/ews-build/steps.py

```diff
@@ -71,32 +72,24 @@ def getHeaders(self):
         return self._get(self.headers)
 
 
-class GitHubMixin(object):
-    def pr_url(self, pr_number=None):
-        pr_number = pr_number or self.get_pull_request_number()
-        if not pr_number:
-            return ''
-        return '{}/pull/{}'.format(self.getProperty('repository', '-'), pr_number)
-
-    def get_pull_request_number(self):
-        pr_number = self.getProperty('pull_request')
-        if pr_number:
-            return int(pr_number)
+class GitHub(object):
+    @classmethod
+    def repository_urls(cls):
+        return [GITHUB_URL + project for project in GITHUB_PROJECTS]
 
-        if self.getProperty('event') != 'pull_request':
-            return None
-        if self.getProperty('repository') not in GITHUB_REPOSITORIES:
-            return None
+    @classmethod
+    def pr_url(cls, pr_number, repository_url=None):
+        if not repository_url:
+            repository_url = '{}{}'.format(GITHUB_URL, GITHUB_PROJECTS[0])
 
-        match = BRANCH_PR_RE.match(self.getProperty('branch', ''))
-        if not match:
-            return None
-        pr_number = int(match.group('id'))
-        self.setProperty('pull_request', pr_number)
-        return pr_number
+        if repository_url not in GitHub.repository_urls():
+            return ''
+        if not pr_number or isinstance(pr_number, int):
```

Comment:
This if condition seems incorrect, please use explicit parenthesis to make it correct and clear.